### PR TITLE
Supporters now uses new collections endpoint

### PR DESCRIPTION
**Pull request by DarrenN**
**To test:**
In the Supporter example, set `apiVersion` to `2016-08-08` in constructor and `apiUrl` to dev

**Lantern schemas:**
```bash
curl -X GET \
     -H "Gw-Api-Version: 2016-08-08" \
     -H "Authorization: Basic cHViLWxhbnRlcm4uZXZlbnRzLWhvcml6b24tLU5YTzZjcmVNV2IwR0JwMXhKbm9tSEVMeURyMHJLQjAzTEFWaE9JaE9nVmZvVlpSMEpCazFDRTJzYWx3M3MyLnpiTk5LLlJPN1VKX3RGQzNBaWZiT2x3Og==" \
     -H "Cache-Control: no-cache" \
     -H "Postman-Token: daa83e87-1eab-2128-4f57-c61328099756" \
     "https://api.dev.thegroundwork.com/collections/supporters/schemas/"
```

### CODE REVIEW

**Comment by DarrenN** [comment]
@damassi this migrates the Supporter service to the new Collections endpoint. Would like to get initial hot takes on design.

**Comment by damassi** [comment]
@DarrenN - I've been trying to figure out how to run the create from the existing supporters example for a while but it looks like the example hasn't been updated (need to pass in `schemaId` to `gw.supporters.create`) and when I trace out errors it looks like additional params are being passed. I don't know enough about walrus to debug, so we can hop on tomorrow. 

In the meantime, can you give me instructions on how you've been testing? Seems like you're working from the console rather than the actual example.

File: src/Supporter.js [hunk 1]
```diff
@@ -122,6 +120,7 @@ export default class Supporter extends Collection {
    */
   create(schemaId, form = {}) {
     // Return a mock error response with validation errors
+    console.log(form)
```
Comment (DarrenN) [inline src/Supporter.js]: wat

**Comment by DarrenN** [comment]
@damassi in `supporter.ejs` - when you instantiate gw on like 142:

```javascript
var gw = new Groundwork({
  apiUrl: '... dev ...',
  apiVersion: '2016-08-08'
});
```

Use an API key from a newer org and you should be good to go.

**Comment by DarrenN** [comment]
Also, I have a general aversion to using `null` in method signatures. So I'm thinking of changing `Supporter.create` from:

```javascript
gw.supporters.create(schemaId, {form}); // with schemaId
gw.supporters.create(null, {form}); // without schemaId - yuck
```

To a curried form like:

```javascript
gw.supporters.create(schemaId)({form}); // with schemaId
gw.supporters.create()({form}); // without schemaId

// alternates
const p = gw.supporters.create(schemaId);
p({form});

const p = gw.supporters.create();
p({form});

// Would let you do craziness like
const p = gw.supporters.create(schemaId);
[xs].forEach(p);
```

How do you feel about that from a design / developer experience perspective?

**Comment by damassi** [comment]
Hey Darren -- I def like the curried version better because I can't stand seeing `null` arguments passed into the first place followed by options. The other option I was thinking -- and this seems like it would be preferable in terms of simplicity unless you expect the end user to take advantage of everything a curried fn can offer (iteration, front-loading, etc) -- is passing the schemaId into the second `form` object and then just running `omit(['schemaId'], form)` on the inside before POST.

**Comment by damassi** [comment]
Another reason why I think passing it into the form object is better is because it keeps things inline with the rest of our method signatures.

**Comment by DarrenN** [comment]
> is passing the schemaId into the second form object and then just running omit(['schemaId'], form) on the inside before POST.

Ah interesting idea. Lemme mull it over.

**Comment by DarrenN** [comment]
@damassi - updated to go with a `schemaId` field within the `form` object itself. I would have preferred using the curried function, but as you noted this option keeps us consistent with other methods in the lib. Customers can also drop the `schemaId` into a hidden form field.

**Comment by damassi** [comment]
Pulled down, ran everything and works great 👍

**Pull request merged**